src/Service/Services/MetadataProviders/CosmosSqlMetadataProvider.cs

```diff
@@ -225,15 +228,9 @@ public List<string> GetSchemaGraphQLFieldNamesForEntityName(string entityName)
         /// <returns></returns>
         public string? GetSchemaGraphQLFieldTypeFromFieldName(string entityName, string fieldName)
         {
-            List<FieldDefinitionNode>? fields;
-
-            // Check if entity name is using alias name, if so, fetch graph type name with the entity alias name
-            foreach (string graphQLType in _graphQLSingularTypeToEntityNameMap.Keys)
+            if (_graphQLTypeToFieldsMap.TryGetValue(entityName, out List<FieldDefinitionNode>? fields))
```

Comment:
Same - entityName may not match any graphQLType

Reply:
I'll defer to the comment I made on [this comment](https://github.com/Azure/data-api-builder/pull/1402#discussion_r1244505927).